### PR TITLE
www/nginx: Add Possibility to Disable User Agent Based Blocks

### PR DESCRIPTION
**Pull request by fzoske**
Hi,

some of my tools (using python urllib) need to access a WebServer behind OpnSense NGINX, but it's not possible, as they "perma" blocked by a hard coded if statement.

This PR adds the possibility to disable this User Agent Based Blocks entirely for a http server.

Best regards,
Fabian

### CODE REVIEW

**Comment by fabianfrz** [comment]
Hi, your pull request looks ok for me but this are the no 1 bots in my log (probably because it is installed on many maches which have been hacked) They commonly scan for Wordpress logins, PHPMyAdmin and simmilar tools.

**Comment by fzoske** [comment]
Hi,

you're certainly right, but often legit tools get misused (e.g. CURL or Python urllib).
They're not all bad and there should be a way to unblock one or all of them.
That's why I let the block enabled and added the (advanced) option to disable it.

If you're not happy with this approach, another could be a multiselect dropdown where you would unblock individual ones.

Best regards,
Fabian

File: www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/forms/httpserver.xml [hunk 1]
```diff
@@ -79,6 +79,13 @@
     <advanced>true</advanced>
     <help>Blocks files like .htaccess files or other files not intended for the public.</help>
   </field>
+  <field>
+    <id>httpserver.enable_default_user_agent_block</id>
+    <label>Enable User Agent Based Blocks</label>
```
Comment (fabianfrz) [inline www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/forms/httpserver.xml]:
```suggestion
    <label>Enable Bot Protection</label>
```

File: www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/forms/httpserver.xml [hunk 2]
```diff
@@ -79,6 +79,13 @@
...
+    <type>checkbox</type>
+    <advanced>true</advanced>
+    <help>Blocks some User Agents like Ronin/Nmap/Python-Urllib, etc.</help>
```
Comment (fabianfrz) [inline www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/forms/httpserver.xml]:
```suggestion
    <help>Blocks the request when a possibly bad bot is detected and adds the originating IP to the managed firewall alias for permanent blocking.</help>
```

**Comment by fabianfrz** [comment]
please also don't forget a version bump (makefile) and the release note (pkg-descr)

**Comment by fabianfrz** [comment]
I added the release note in #930 to my changes. So I'll probably add it in version **1.4** of the plugin.

**Comment by fzoske** [comment]
I've changed the name as I like yours better

**Comment by fabianfrz** [comment]
@fzoske just wait for the approval by @fichtner for merge.

File: www/nginx/src/opnsense/service/templates/OPNsense/Nginx/http.conf [hunk 3]
```diff
@@ -130,6 +130,7 @@ server {
         root /var/etc/acme-client/challenges;
     }
 {% endif %}
+{% if server.disable_bot_protection is defined and server.disable_bot_protection == '0' %}
```
Comment (fabianfrz) [inline www/nginx/src/opnsense/service/templates/OPNsense/Nginx/http.conf]: can you try that one - should be more fail safe
```suggestion
{% if server.disable_bot_protection is not defined or server.disable_bot_protection != '1' %}
```

Comment (fabianfrz) [reply www/nginx/src/opnsense/service/templates/OPNsense/Nginx/http.conf]: yours would currently suppress the block if the option is missing.

Comment (fzoske) [reply www/nginx/src/opnsense/service/templates/OPNsense/Nginx/http.conf]: Sure

**Comment by fabianfrz** [comment]
@fzoske you can delete your branch now.

**Pull request merged**